docs/contracts.rst

```diff
@@ -1406,24 +1406,27 @@ Using For
 
 The directive ``using A for B;`` can be used to attach library
 functions (from the library ``A``) to any type (``B``).
-These functions will receive the object they are called on
-as their first parameter (like the ``self`` variable in
-Python).
 
 The effect of ``using A for *;`` is that the functions from
-the library ``A`` are attached to any type.
+the library ``A`` are attached to *any* type.
 
-In both situations, all functions, even those where the
-type of the first parameter does not match the type of
-the object, are attached. The type is checked at the
+By including a module, its data types including library functions are
+available without having to add further code.
+
+In both situations, *all* functions in the library are attached,
+even those where the type of the first parameter does not
```

Comment:
I think the explanatory sentence about what "attaching a function" actually means ("When such functions are called, ...") has to precede at least this paragraph, otherwise why do we care about the first parameter?

Reply:
Agreed. So changed.